### ISSUE TITLE
Bug at Remove-DbaAgentJobStep since no sqlCredential is passed across and cmdlet fails to execute

### ISSUE THREAD

**Issue by marioanton** [sqlcollaborative/dbatools#3830]
### Before submitting a bug report:

- [x] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [x] Verified this bug is not already reported in an issue
- [x] Verified errors are not related to permissions
- [x] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

## Steps to Reproduce

```sql
/*
    Any T-SQL commands involved or used to produce test objects/data.
*/
```
New-DbaSqlAgentSteps -Settings $Settings -JobName $JobName -Credential $Credential -SqlServer $SqlServer

```powershell
<#
Remove-DbaAgentJobStep : The following exception occurred while trying to 
July 20th 2018 12:06:12Error
enumerate the collection: "An exception occurred while executing a 
July 20th 2018 12:06:12Error
Transact-SQL statement or batch.".
July 20th 2018 12:06:12Error
At C:\bede.dba\main\vendor\dbatools\allcommands.ps1:55495 char:29
July 20th 2018 12:06:12Error
+ ...             Remove-DbaAgentJobStep -SqlInstance $instance -Job $curre ...
July 20th 2018 12:06:12Error
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
July 20th 2018 12:06:12Error
    + CategoryInfo          : NotSpecified: (:) [Remove-DbaAgentJobStep], Exte 
July 20th 2018 12:06:12Error
   ndedTypeSystemException
July 20th 2018 12:06:12Error
    + FullyQualifiedErrorId : ExceptionInGetEnumerator,Remove-DbaAgentJobStep
July 20th 2018 12:06:12Error
July 20th 2018 12:06:12Fatal
The remote script failed with exit code 1#>
```

## Expected Behavior

No errors.

## Actual Behavior

When running the command Remove-DbaAgentJobStep contained in allcommands.ps1 script we get failure due to lack of permissions when the sqlcredential is even passed across. We are hitting this command by initially triggering this 
New-DbaSqlAgentSteps -Settings $Settings -JobName $JobName -Credential $Credential -SqlServer $SqlServer

## Environmental data

 - PowerShell: 

PSVersion                      5.0.10586.117
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.10586.117
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

 - SQL Server: 

 Microsoft SQL Server 2016 (SP1-CU7-GDR) (KB4057119) - 13.0.4466.4 (X64)   Dec 22 2017 11:25:00   Copyright (c) Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2012 R2 Datacenter 6.3 <X64> (Build 9600: ) (Hypervisor) 

This issue would be sorted by adding `-SqlCredential $SqlCredential` to the Remove-DbaAgentJobStep call

**Comment by marioanton** [comment]
Got the fix in place. Will be raising a PR soon

**Comment by niphlod** [comment]
fixed via #3850
